### PR TITLE
Asynchronous backing PR

### PR DESCRIPTION
**Pull request by slumber**
This is the Cumulus companion to asynchronous backing.

There are 2 major changes introduced in this PR, which will be expanded upon in a blog post:

**Consensus Hooks**

The major change of this PR is to introduce a `ConsensusHook` to the `pallet_parachain_system::Config` of a parachain. This `ConsensusHook` is a pluggable component which allows a chain to manage its asynchronous backing backlog. It is enabled by adding the `parameterized-consensus-hook` feature to `pallet-parachain-system` in your runtime. There is an implementation of this for Aura within `pallet-aura-ext`.

**Lookahead Collator for Aura**

Cumulus consensus architecture has been overhauled, and the new "Lookahead" collator logic for Aura takes into account asynchronous backing. Further details on how to integrate this collator will follow in a dedicated post.

See https://github.com/paritytech/polkadot/issues/3779, https://github.com/paritytech/polkadot/pull/5022

### CODE REVIEW

File: client/relay-chain-minimal-node/src/collator_overseer.rs
Comment (mrcnski) [inline client/relay-chain-minimal-node/src/collator_overseer.rs:66]:
```suggestion
	/// Receiver for collation request protocol vstaging.
	pub collation_req_receiver_vstaging:
```

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/3159454

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/3174507

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/3224576

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/3232372

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/3269669

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/3287744

File: pallets/parachain-system/src/validate_block/implementation.rs
Comment (rphmeier) [inline pallets/parachain-system/src/validate_block/implementation.rs:178]: @slumber just revisiting why this was removed

File: parachains/runtimes/assets/asset-hub-kusama/src/lib.rs
Comment (rphmeier) [inline parachains/runtimes/assets/asset-hub-kusama/src/lib.rs:1237]: @slumber Unfortunately I don't think we can safely expect all downstream users to enact this same change for the moment.

Comment (rphmeier) [reply parachains/runtimes/assets/asset-hub-kusama/src/lib.rs:1237]: We should restore the `CheckInherents` logic to `register_validate_block!` but not use it in any runtimes that have the `aura_ext::FixedVelocityConsensusHook`.

@BradleyOlson64 this is something for the update guide.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/3394622

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/3403157

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/3403318

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/3409500

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/3409625

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/3410014

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/3410130

File: bridges/bin/runtime-common/Cargo.toml
Comment (liamaharon) [inline bridges/bin/runtime-common/Cargo.toml:45]: I think these need to be updated back to `master`? 

Maybe you're already aware and chose not to, but you can use [diener](https://github.com/paritytech/diener) to patch a local checkout of polkadot. 

You may find it easier than changing the branch in all these cargo.tomls.

```bash
# in polkadot
git checkout rh-async-backing-feature
cd ../cumulus
diener patch --crates-to-patch ../polkadot --polkadot
```

Comment (slumber) [reply bridges/bin/runtime-common/Cargo.toml:45]: It was updated to `rh-async-backing` using diener, and will be reverted to master before the merge.

(imo it was more convenient to point to origin, instead of local branch, so you're not locking polkadot folder)

File: parachains/integration-tests/emulated/collectives/collectives-polkadot/src/tests/fellowship.rs
Comment (ordian) [inline parachains/integration-tests/emulated/collectives/collectives-polkadot/src/tests/fellowship.rs:22]: why comment out if you put `ignore` attribute? is it not compiling?

**Review by ordian — APPROVED** [review]
Haven't reviewed deeply, but approving to unblock merge.

Don't forget to revert diener patches.

File: parachains/integration-tests/emulated/collectives/collectives-polkadot/src/tests/fellowship.rs
Comment (slumber) [reply parachains/integration-tests/emulated/collectives/collectives-polkadot/src/tests/fellowship.rs:22]: CI executes ignored tests

**Review by eskimor — APPROVED** [review]

**Review by bkontur — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/3411632

**Review by liamaharon — APPROVED** [review]

**Review by acatangiu — APPROVED** [review]

**Review by sandreim — APPROVED** [review]

**Review by rphmeier — APPROVED** [review]

File: Cargo.lock
Comment (ordian) [inline Cargo.lock:9974]: these versions should be bumped in polkadot

**Comment by ordian** [comment]
bot merge

**Comment by muharem** [comment]
@slumber any reason why the integration tests were commented? plans to get them back into master?

**Comment by slumber** [comment]
> @slumber any reason why the integration tests were commented? plans to get them back into master?

See https://github.com/paritytech/cumulus/issues/3027 https://github.com/paritytech/cumulus/issues/2703

They're commented out because CI executes tests marked with #[ignore] anyway

**Pull request merged**
